manifests/server/config_entry.pp

```diff
@@ -9,15 +9,8 @@
   Enum['present', 'absent']                               $ensure = 'present',
   String[1]                                               $key    = $name,
   Optional[Variant[String[1], Numeric, Array[String[1]]]] $value  = undef,
-  Variant[Boolean, String[1]]                             $path   = false
+  Variant[String[1], Stdlib::Absolutepath]                $path   = $postgresql::server::postgresql_conf_path
```

Comment:
Why `Stdlib::Absolutepath` if `String[1]` is also accepted? Then I'd simplify it to just `String[1]`. Having said that: does a relative path ever work or should it be simplified to an absolute path?

Reply:
Agreed. Thinking about it `Stdlib::Absolutepath` probably makes more sense. The `pick()` idea is nice, but I think the current implementation is easier to read.

Reply:
> The `pick()` idea is nice, but I think the current implementation is easier to read.

In theory it allows you to stop caring about the compile order, if you also add `include postgresql::server` but that's not there today either.

Reply:
Regarding the Data Type I just took the Type from the `postgresql::server` class. I would suggest changing this in the whole module is part of #1467. I am happy to create  a PR for that.

Reply:
PR ist open #1491